src/packages/client/src/generation/TSClient/TSClient.ts

```diff
@@ -81,7 +81,7 @@ export class TSClient implements Generatable {
     const code = `${commonCodeJS({ ...this.options, browser: false })}
 
 const dirnamePolyfill = path.join(process.cwd(), ${JSON.stringify(cwdDirname)})
-const dirname = __dirname.length === 1 ? dirnamePolyfill : __dirname
+const dirname = (__dirname.includes('.next/serverless') || __dirname === '/') ? dirnamePolyfill : __dirname
```

Comment:
Do we usually have framework specific things in this file? 
This feels like a very specialized solution for a more generic problem.

Reply:
Yes, it's quite common, unfortunately. We have `if (netlify)` checks in some places too.

Reply:
The `dirnamePolyfill` was specifically made for `next` I think, some comment seems to point into this direction. I agree with you that this is not generic. I wanted to generalize the fix by using `process.cwd()` instead but it breaks many of our existing tests.

Because we cannot predict the state of the client (bundled or not, or in a different folder, in a build step or prod), it makes it difficult to use `__dirname`. I get the feeling that it might be a bad idea in this case. It would be easier to look relatively to the process rather than relatively to the file (we write the relative paths in the `config` variable in the generated client. Once bundled and moved around, those paths aren't valid anymore).

Reply:
I would expect things to be searched (binaries, schemas) relatively to the process unless a custom `output` path has been provided.